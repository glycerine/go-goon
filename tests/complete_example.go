package main

import "github.com/shurcooL/go-goon"

func main() {
	goon.Dump(map[string]int64{
		"x": 1,
		"y": 4,
		"z": 7,
	})

	goon.Dump([]int32{1, 5, 8})

	{
		x := (*string)(nil)
		goon.Dump(x, nil)
	}

	goon.Dump([]byte("foodboohbingbongstrike123"))

	goon.Dump(uintptr(0), uintptr(123))

	{
		f := func() { println("This is a func.") }

		goon.Dump(f)

		f2 := func(a int, b int) int {
			c := a + b
			return c
		}

		goon.Dump(f2)

		unexportedFuncStruct := struct {
			unexportedFunc func() string
		}{func() string { return "This is the source of an unexported struct field." }}

		goon.Dump(unexportedFuncStruct)
	}
}
