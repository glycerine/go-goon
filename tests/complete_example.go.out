(map[string]int64)(map[string]int64{
	(string)("x"): (int64)(1),
	(string)("y"): (int64)(4),
	(string)("z"): (int64)(7),
})
([]int32)([]int32{
	(int32)(1),
	(int32)(5),
	(int32)(8),
})
(*string)(nil)
(interface{})(nil)
([]uint8)([]uint8{
	(uint8)(102),
	(uint8)(111),
	(uint8)(111),
	(uint8)(100),
	(uint8)(98),
	(uint8)(111),
	(uint8)(111),
	(uint8)(104),
	(uint8)(98),
	(uint8)(105),
	(uint8)(110),
	(uint8)(103),
	(uint8)(98),
	(uint8)(111),
	(uint8)(110),
	(uint8)(103),
	(uint8)(115),
	(uint8)(116),
	(uint8)(114),
	(uint8)(105),
	(uint8)(107),
	(uint8)(101),
	(uint8)(49),
	(uint8)(50),
	(uint8)(51),
})
(uintptr)(nil)
(uintptr)(0x7b)
(func())(0x2570)
(func(int, int) int)(0x25b0)
